weed/s3api/s3api_object_multipart_handlers.go

```diff
@@ -201,8 +202,9 @@ func (s3a *S3ApiServer) PutObjectPartHandler(w http.ResponseWriter, r *http.Requ
 	}
 	defer dataReader.Close()
 
-	uploadUrl := fmt.Sprintf("http://%s%s/%s/%04d.part?collection=%s",
-		s3a.option.Filer, s3a.genUploadsFolder(bucket), uploadID, partID, bucket)
+	maxMB := int(math.Ceil(float64(r.ContentLength) / 1024 / 1024))
+	uploadUrl := fmt.Sprintf("http://%s%s/%s/%04d.part?collection=%s&maxMB=%d",
```

Comment:
S3 upload part can be fairly big. Why not leave the filer to determine the chunk size?

Reply:
The size of the downloadable part of S3 is convenient for the user himself, but you can limit it to a reasonable size, for example, filer maxMB * 5

Reply:
I prefer not to change here. Filer can handle the chunking transparently. S3 server should not need to worry about it.

Reply:
This is necessary in special cases, the maxMB code is on average less than the size of a s3 chunk, as a result of which several chunkviews must be delivered for one c3 request.
on average, the size of a s3 chunk is from 5mb to 50mb,  so i configured maxMB to 64.

Reply:
but why need to have a large chunk instead of several smaller ones? The latter is more efficient to process.

Reply:
> but why need to have a large chunk instead of several smaller ones? The latter is more efficient to process.

It depends on the content, if I have a regular 100Gb source file I want the chunk size of at least 100 mb.

Reply:
Filer has a feature to merge every 1000 chunk ids into one chunk, and just reference that chunk id.

Reply:
@kmlebedev

> It depends on the content, if I have a regular 100Gb source file I want the chunk size of at least 100 mb.

full chunk goes in memory before it can be processed, so you will be limited with these big chunks, there's no streaming afaik.